src/NServiceBus.Testing/ExpectedInvocations/ExpectReplyToOriginator.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace NServiceBus.Testing.ExpectedInvocations
+{
+    using System;
+    using System.Linq;
+
+    class ExpectReplyToOriginator<TMessage> : ExpectInvocation
+    {
+        public ExpectReplyToOriginator(Func<TMessage, bool> check = null)
+        {
+            this.check = check ?? (m => true);
+        }
+
+        public override void Validate(TestableMessageHandlerContext context)
+        {
+            var repliedMessages = context.RepliedMessages
+                .Where(i => i.Message.GetType() == typeof(TMessage))
```

Comment:
I think this should be changed to the `Containing<TMessage>` extension method...